galsim/config/image.py

```diff
@@ -242,27 +194,67 @@ def BuildImage(config, logger=None, image_num=0, obj_num=0,
     SetupConfigImageNum(config,image_num,obj_num)
 
     image_type = config['image']['type']
-    if image_type not in valid_image_types:
-        raise AttributeError("Invalid image.type=%s."%image_type)
 
-    build_func = eval(valid_image_types[image_type][0])
-    all_images = build_func(
-            config=config, logger=logger,
-            image_num=image_num, obj_num=obj_num,
-            make_psf_image=make_psf_image, 
-            make_weight_image=make_weight_image,
-            make_badpix_image=make_badpix_image)
+    # Build the rng to use at the image level.
+    seed = galsim.config.SetupConfigRNG(config)
+    if logger and logger.isEnabledFor(logging.DEBUG):
+        logger.debug('image %d: seed = %d',image_num,seed)
+
+    # Do the necessary initial setup for this image type.
+    setup_func = valid_image_types[image_type]['setup']
+    xsize, ysize = setup_func(config, image_num, obj_num, image_ignore, logger)
+
+    # Given this image size (which may be 0,0, in which case it will be set automatically later),
+    # do some basic calculations
+    SetupConfigImageSize(config,xsize,ysize)
+    if logger and logger.isEnabledFor(logging.DEBUG):
+        logger.debug('image %d: image_size = %d, %d',image_num,xsize,ysize)
+        logger.debug('image %d: image_origin = %s',image_num,config['image_origin'])
+        logger.debug('image %d: image_center = %s',image_num,config['image_center'])
+
+    # Sometimes an input field needs to do something special at the start of an image.
+    galsim.config.SetupInputsForImage(config,logger)
+
+    # Likewise for the extra output items.
+    galsim.config.SetupExtraOutputsForImage(config,logger)
+
+    # Actually build the image now.  This is the main working part of this function.
+    # It calls out to the appropriate build function for this image type.
+    build_func = valid_image_types[image_type]['build']
+    image = build_func(config, image_num, obj_num, logger)
+
+    # Store the current image in the base-level config for reference
+    config['current_image'] = image
+
+    # Mark that we are no longer doing a single galaxy by deleting image_pos from config top
+    # level, so it cannot be used for things like wcs.pixelArea(image_pos).
+    if 'image_pos' in config: del config['image_pos']
```

Comment:
I'm sure isn't important from a performance standpoint, but you can write this "delete if exists" operation more simply and with one less lookup as `config.pop("image_pos", None)`.